src/main/java/Problem1/ValidParentheses.java

```diff
@@ -4,8 +4,54 @@
 
     // Do not change signature (function name, parameters, return type)
     public static boolean isValid(String str) {
-        // homework
-        // !!! must use ArrayStack or LinkedListStack from problem 1
-        return false; // place holder
+
+        LinkedListStack<Character> data = new LinkedListStack<Character>();
+
+        if (str == null)
+        {
+            return true;
+        }
+        if (str.length() % 2 != 0) {
+            return false;
+        }
+
+        for (int i = 0; i < str.length(); i++)
+        {
+            if (str.charAt(i) == '(' || str.charAt(i) == '{' || str.charAt(i) == '[')
+            {
+                data.push(str.charAt(i));
+            }
+            else
+                {
+                if (data.size() == 0)
```

Comment:
this is two level of condition checks. an if within the else of another if. this tends to be over complicated and hard to catch error. let's try to refactor this into a single level of condition check.